Assets/Scripts/PowerUps/PowerUp.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public abstract class PowerUp : MonoBehaviour, IInteractable
+{
+    protected void PlayPickUpSound() 
+    {
+        AudioManager.m_Instance.PlaySound("MAGIC_Powerup", 1.0f, 1.2f);
+    }
+
+    public void Interact(ICanInteract interactor) 
```

Comment:
I think the interact method should contain all the common behaviour called by all powerup classes, but to implement specific behaviour for each powerup it can call a virtual method that has not been implemented.

Reply:
Hmmm... in that case I'll have another abstract method to handle specific behavior